pkg/common/utils.go

```diff
@@ -115,6 +123,56 @@ func ConvertSITimeout(millis int64) time.Duration {
 	return time.Duration(result)
 }
 
+// ConvertSITimeoutWithAdjustment Similar to ConvertSITimeout, but this function also adjusts the timeout if
+// "creationTime" is defined. It's used during Yunikorn restart, in order to properly track how long a placeholder pod should
+// be in "Running" state.
+func ConvertSITimeoutWithAdjustment(siApp *si.AddApplicationRequest) time.Duration {
+	result := ConvertSITimeout(siApp.ExecutionTimeoutMilliSeconds)
+	adjusted := adjustTimeout(result, siApp)
+
+	return adjusted
+}
+
+func adjustTimeout(timeout time.Duration, siApp *si.AddApplicationRequest) time.Duration {
+	creationTimeTag := siApp.Tags[interfaceCommon.DomainYuniKorn+CreationTime]
+	if creationTimeTag == "" {
+		return timeout
+	}
+
+	created := ConvertSITimestamp(creationTimeTag)
+	if created == Undefined {
+		return timeout
+	}
+	now := currentTime()
```

Comment:
Testing and all other cases should never need to override `time.Now()`. We do not care if it is exactly 1 or 10 minutes as long as it is less than what is set as the timeout.

Reply:
Fixed